examples/hybrid/schur_complement_W.jl

```diff
@@ -206,3 +198,82 @@ function linsolve!(::Type{Val{:init}}, f, u0; kwargs...)
         end
     end
 end
+
+"""
+    thomas_algorithm!(A, b)
+
+Thomas algorithm for solving a linear system A x = b,
+where A is a tri-diagonal matrix.
+A and b are overwritten.
+Solution is written to b
+"""
+function thomas_algorithm!(A, b)
```

Comment:
Let's move this, and the remaining functions added in this file into `src/`, since we want it useable in ClimaAtmos.

Reply:
Yes, this should be part of ClimaCore itself.